From 644785d73659bdf1108d9e86b7a1d36129de7f81 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 23 May 2021 16:07:17 +0300
Subject: [PATCH] [SW-42722] sec: fwupd: allow fit access on resp

Need to make sure that response's fit content, should be accessible by the
controller in security mode.
The controller-device share area may be blocked as part of its behavior.
When the device send out a response with an additional content. The
share area, where the content located, should be unblocked by the
preboot, allowing access from the host. When sending out a status with
FIT, the shared area gets unblocked. And only blocked again after
receiving RST_STATE event from the controller.

Change-Id: Id40f517f4605e74285d12a92ac5e63de91854cca
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 17bbc60..b483d6b 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -101,6 +101,9 @@
 	struct hl_upd_status status;
 	uintptr_t off = (uintptr_t)offset;
 
+	if (offset)
+		preboot_security_dynamic_img_sram_open();
+
 	status.val = 0;
 	status.status = sts;
 	status.offset = (u32)(off & ~HL_UPD_STS_OFFSET_MASK);
@@ -548,8 +551,14 @@
 		preboot_puts(": default case - err\n");
 	}
 
-	if (err || state->exec_res == NOT_FOUND)
-		module_fwupd_reset_state(state);
+	/* NOTE: at this point we used to have a code that
+	 *       reset the protocol FSM state, on error,
+	 *       and on NOT_FOUND result. As part of the operation,
+	 *       It used to close the host's access to the response memory
+	 *       the reset state at those cases are not a must.
+	 *       and it may close the memory access, before an explicit
+	 *       request for RST_STATE, invoked by the host.
+	 */
 }
 
 void module_fwupd_run(void)
